### PR TITLE
Collect Farm Target: Only report errors for not found addons if no valid farm addon found

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Only report errors for not found addons if no valid farm addon found

## Additional info

Solved redundant error logs for e.g. `royalrender` addon even if the studio isn't using it. Now it only logs errors for not found addons if NO valid farm addon is found at all.

Avoids this error log if it's irrelevant:

image

These error logs were also visibile in screenshots here: https://github.com/ynput/ayon-core/pull/623#pullrequestreview-2109615241

## Testing notes:

1. Publishing to Deadline and Royal Render should still work

**Note** that with this change if both Royal Render and Deadline are enabled addons that now Deadline takes precedence where with the old logic Royal Render would be picked.

### CODE REVIEW

File: client/ayon_core/plugins/publish/collect_farm_target.py [hunk 1]
```diff
@@ -14,22 +14,20 @@ def process(self, instance):
...
-        for farm_renderer in ["deadline", "royalrender"]:
+        farm_renderer_addons = ["deadline", "royalrender"]
+        for farm_renderer in farm_renderer_addons:
             addon = addons_manager.get(farm_renderer, False)
-
-            if not addon:
-                self.log.error("Cannot find AYON addon '{0}'.".format(
-                    farm_renderer))
-            elif addon.enabled:
+            if addon.enabled:
```
Comment (kalisp) [inline client/ayon_core/plugins/publish/collect_farm_target.py:22]: Wouldn't this fail if addon is not found?

Comment (BigRoy) [reply client/ayon_core/plugins/publish/collect_farm_target.py]: You are one hundred percent correct - fixed with https://github.com/ynput/ayon-core/pull/637/commits/3b5791a8e08c8b9831b2ad95a2511006fcf3478c

File: client/ayon_core/plugins/publish/collect_farm_target.py [hunk 2]
```diff
@@ -14,22 +14,20 @@ def process(self, instance):
...
                 farm_name = farm_renderer
-
-        if farm_name:
-            self.log.debug("Collected render target: {0}".format(farm_name))
-            instance.data["toBeRenderedOn"] = farm_name
+                break
         else:
```
[the remark this answers is not in the document]
Comment (BigRoy) [reply client/ayon_core/plugins/publish/collect_farm_target.py:25]: It's a [`for..else`](https://book.pythontips.com/en/latest/for_-_else.html#else-clause)
The beauty of Python.

[the remark this answers is not in the document]
Comment (kalisp) [reply client/ayon_core/plugins/publish/collect_farm_target.py:25]: Omg, this is ugly af. I could understand `for..finally`, but this one is just weird.

**Review by kalisp — APPROVED** [review]
Print is gone

**Pull request merged**